### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Tarasulko**

### CODE REVIEW

File: src/herbivoresAndCarnivores.js [hunk 1]
```diff
@@ -1,15 +1,46 @@
...
+  hide() {
+    if (this.hidden === false) {
+      this.hidden = true;
+
+      return;
+    }
+
+    if (this.hidden === true) {
+      this.hidden = false;
+    }
```
Comment (aliceblunt8) [inline src/herbivoresAndCarnivores.js:28]:
```suggestion
    this.hidden = !this.hidden;
```

File: src/herbivoresAndCarnivores.js [hunk 2]
```diff
@@ -1,15 +1,46 @@
...
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Carnivore || animal.hidden === true) {
+      return;
+    }
+
+    animal.health -= 50;
```
Comment (aliceblunt8) [inline src/herbivoresAndCarnivores.js:38]:
```suggestion
    if (animal instanceof Herbivore && !animal.hidden) {
      animal.health -= 50;
    }
```

File: src/herbivoresAndCarnivores.js [hunk 3]
```diff
@@ -1,15 +1,46 @@
...
+
+    if (animal.health <= 0) {
+      Animal.alive = [...Animal.alive.filter(a => a.name !== animal.name)];
```
Comment (aliceblunt8) [inline src/herbivoresAndCarnivores.js:41]: what return filter? new or old array?

**Review by aliceblunt8 — CHANGES_REQUESTED** [review]

File: src/herbivoresAndCarnivores.js (hunk 3 above)
Comment (Tarasulko) [reply src/herbivoresAndCarnivores.js]: fixed

**Review by aliceblunt8 — APPROVED** [review]

**Pull request closed without merging**